contracts/staking/DefaultTargetDispenserL2.sol

```diff
@@ -393,6 +405,49 @@ abstract contract DefaultTargetDispenserL2 is IBridgeErrors {
         _locked = 1;
     }
 
+    /// @dev Migrates funds to a new specified L2 target dispenser contract address.
+    /// @notice The contract must be paused to prevent other interactions.
+    /// @notice Owner will be zeroed and the contract becomes paused for good.
+    function migrate(address newL2TargetDispenser) external {
```

Comment:
This function migrates all the funds to the new L2 target dispenser, and freezes all the contract activity.

Reply:
@DavidMinarsch @mariapiamo this has been implemented to address a preliminary discussion stating that if anything goes wrong on L1, the funds are stuck in this contract forever. This proposed implementation allows to deploy another L2 contract and migrate all the funds there.

Reply:
Good! Even if the DAO would always be able to "drain" eventual withheld tokens with new staking contract, this removes completely the need of manual interaction for "fund recovery".